src/Cryptol/ModuleSystem/Renamer.hs

```diff
@@ -161,8 +161,10 @@ renameModule' thisNested env mpath m =
          allImps   = openLoop allNested env openDs imps
 
      (inScope,decls') <-
-        shadowNames allImps $
```

Comment:
This was basically the source of the spurious overlapping errors, right?

Reply:
In short yes, because we were checking that different imports don't define the same thing, but is actually OK if they do as long as we don't use.  Btw, I find the input argument to `shadowNames` to be quite confusing, and we may want to change in a future refactoring.